src/rosdiscover/observer/observer.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+__all__ = ("Observer",)
+
+import contextlib
+from abc import ABC, abstractmethod
+import typing
+from typing import Dict, Iterator
+
+import roswire
+from roswire import AppInstance, ROSVersion
+
+from ..interpreter import NodeContext, SystemSummary
+
+if typing.TYPE_CHECKING:
+    from ..config import Config
+
+
+class Observer(ABC):
+
+    @classmethod
+    @contextlib.contextmanager
```

Comment:
There doesn't seem to be anything gained by using a context manager here? (I.e., there are no `try`/`finally` blocks, and so a context manager isn't necessary).

Reply:
Perhaps I'm not understanding this properly, but line 79 of cli.py has this wrapped in a `with`

Reply:
Done